detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/Project/ProjectXmlResolver.cs

```diff
@@ -87,6 +87,25 @@ public DependencyResult Process()
                             var dep = new NugetDependency(include.Value, NuGet.Versioning.VersionRange.Parse(version.Value));
                             tree.Add(dep);
                         }
+                        if (version == null)
+                        {
+                            String versionStr = null;
+
+                            foreach (XmlNode node in package.ChildNodes)
+                            {
+                                if (node.Name == "Version")
+                                {
+                                    versionStr = node.InnerXml;
+                                    break;
+                                }
+                            }
+
+                            if (include != null && !String.IsNullOrWhiteSpace(versionStr))
+                            {
+                                var dep = new NugetDependency(include.Value, NuGet.Versioning.VersionRange.Parse(versionStr));
+                                tree.Add(dep);
+                            }
+                        }
```

Comment:
This if block on 103 and the one on 85 look very similar to me. Is there anyway to set version in whichever way works and then  just do this block once?

Reply:
Yeah certainly good point.